driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java

```diff
@@ -281,7 +279,7 @@ private boolean handleReplicaSetMemberChanged(final ServerDescription newDescrip
                                 newDescription.getSetVersion(), newDescription.getElectionId(),
                                 maxSetVersion, maxElectionId));
                     }
-                    addressToServerTupleMap.get(newDescription.getAddress()).server.invalidate();
+                    addressToServerTupleMap.get(newDescription.getAddress()).server.resetToConnecting();
```

Comment:
According to the `updateRSFromPrimary` subroutine in SDAM, this path should only `replace description with a default ServerDescription of type "Unknown"`

Reply:
Should `resetToConnecting` save the `newDescription`'s topologyVersion? Does it matter?

Reply:
I don't think it matters, but if you think it does it should be in the spec.

Reply:
Agreed. Please ignore.